scripts/docker-helper/cht-docker-compose.sh

```diff
@@ -251,6 +278,90 @@ get_load_avg() {
   fi
 }
 
+get_container_process_count(){
+  container=$1
+  echo $(docker top "${container}" | tail -n +2 | wc -l)
+}
+
+log_iteration(){
+  counter=$1
+  reboot_count=$2
+  last_msg=$(echo "$3" | tr -d '"')
+  docker_call=$4
+  line_head="$(date) pid=\"$$\" count=\"${counter}\""
+  load_now=$(echo $(get_load_avg)|cut -d" " -f 1)
+
+  # output the log in the same directory as the env file
+  log_location=$(dirname $envFile)
+  logname="${log_location}/cht-docker-compose.log"
+
+  full_url=$(get_local_ip_url $lanAddress)
+  portIsOpen=$(port_open "$lanAddress" "$CHT_HTTPS")
+  if [ "$portIsOpen" = "0" ]; then
+    port_status='open'
+    http_code=$(curl -k --silent --show-error --head "$full_url" --write-out '%{http_code}' | tail -n1)
+    ssl_verify=$(curl -k --silent --show-error --head "$full_url" --write-out '%{ssl_verify_result}' | tail -n1)
+    if [ "$ssl_verify" = "0" ]; then
+      ssl_verify=yes
+    else
+      ssl_verify=no
+    fi
+  else
+    port_status='closed'
+    http_code='NA'
+    ssl_verify='no'
+  fi
+
+  if [ $counter -eq 1 ]; then
+
+    echo "${line_head} \
+item=\"start\" \
+URL=\"$full_url\" \
+IP=\"$lanAddress\" \
+port_https=\"$CHT_HTTPS\" \
+port_http=\"$CHT_HTTP\" \
+project_name=\"$COMPOSE_PROJECT_NAME\" \
+total_containers=\"$(get_global_running_container_count)\"\
+" >& 1 >> $logname
+
+  fi
+
+  container_stat=''
+  IFS=' ' read -ra containersArray <<<"$ALL_CONTAINERS"
+  for container in "${containersArray[@]}"; do
+    RUNNING=$(docker inspect --format="{{.State.Running}}" "${container}" 2> /dev/null)
+    if [ "$RUNNING" == "true" ]; then
+      logs=$(docker logs -n1 ${container})
+      logs=$(echo $logs | tr -d '"')
+
+      # todo - maybe grab horti logs if container ends in "_medic-os_1"?
+      # docker exec -it helper_test_medic-os_1  tail -n3 /srv/storage/horticulturalist/logs/horticulturalist.lo
+    else
+      RUNNING="false"
+      logs="NA"
+    fi
+
+    echo "${line_head} item=\"docker_logs\" container=\"${container}\" processes=\"$(get_container_process_count ${container})\" last_log=\"$logs\"" >& 1 >> $logname
+    container_stat="${container}=\"${RUNNING}\" ${container_stat}"
+  done
+
+  echo "${line_head} \
+item=\"status\" \
+CHT_count=\"$(get_running_container_count "$ALL_CONTAINERS")\" \
+port_stat=\"$port_status\" \
+http_code=\"$http_code\" \
+ssl_verify=\"$ssl_verify\" \
+reboot_count=\"$reboot_count\" \
+docker_call=\"$docker_call\" \
+last_msg=\"$last_msg\" \
+load_now=\"$load_now\" \
+$container_stat\
+" >& 1 >> $logname
```

Comment:
A quoted key value pair was my base for being easy to parse and easy to digest for humans and computers alike.  Admittedly, there's not much difference here between this and JSON! However, since this is a bit more legible to humans than JSON due to less braces and commas, I say let's stick with what we have and change at a later date if we find JSON would be more helpful.

Thanks for the inquiry!